controllers/product.go

```diff
@@ -159,5 +159,47 @@ func (ctrl ProductController) DeleteOne(c *gin.Context) {
     c.IndentedJSON(http.StatusOK, gin.H{"success": true})
   }
 
-  return
+}
+
+func (ctrl ProductController) EditOne(c *gin.Context) {
+
+  id := c.Param("id")
+  productId, err := strconv.ParseInt(id, 10, 64)
+  if productId == 0 || err != nil {
+    c.AbortWithStatusJSON(http.StatusBadRequest, errors.ErrInvalidParameter)
+    return
+  }
+
+  var input models.EditProductInput
+  err = c.ShouldBind(&input)
+  if err != nil {
+    c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+    return
+  }
+
+  var product models.Product
+
+  db.DB.First(&product, productId)
+
+  img_url := ctrl.extractImage(c)
+
+  if len(input.Title) != 0 {
+    product.Title = input.Title
+  }
+
+  if len(input.Content) != 0 {
+    product.Content = input.Content
+  }
+
+  if input.Price != 0 {
+    product.Price = input.Price
+  }
+
+  if len(img_url) != 0 {
+    product.ImageURL = img_url
+  }
+
+  db.DB.Save(&product)
```

Comment:
fetch -> edit -> update
that's a perfect example of race conditions :joy: 
consider update multiple fields https://gorm.io/docs/update.html#Updates-multiple-columns 
or ignore it, no one cares anyway :joy: